File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.ts
function longestConsecutive(nums: number[]): number {
	const N = nums.length,
	    	set = new Set<number>(nums);

  let ans = 0;

	for (let num of set.values()) {
		
    if (!set.has(num - 1)) {
			let curr = num,
				cnt = 1;

			while (set.has(curr + 1)) {
				cnt += 1;
				curr += 1;
			}

			ans = Math.max(ans, cnt);
		}
	}

	return ans;
};

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.java
class Solution {
    
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        int max = 0;
        
        for(int num: nums)
            set.add(num);

        for(int num: nums) {
            
            if(!set.contains(num - 1)) {
                int currNum = num, streak = 1;
                
                while(set.contains(++currNum))
                    streak++;
                
                max = Math.max(streak, max);
            }
        }

        return max;
    }
}